src/controllers/HealthCheckController.ts

```diff
@@ -0,0 +1,69 @@
+import { Connection } from 'mongoose';
+
+export interface IResult {
+  database: {
+    isConnected: boolean;
+    status: number;
+    message: string;
+  };
+  service: {
+    isReady: boolean;
+    status: number;
+    message: string;
+  };
+}
+
+export default class HealthCheckController {
+  constructor(private connection: Connection) {}
+
+  public getServiceStatus(): IResult {
+    return {
+      database: {
+        isConnected: this.isConnected(),
+        status: this.connection.readyState,
+        message: this.databaseStatusMessage(),
+      },
+      service: {
+        isReady: this.isConnected(),
+        status: this.isConnected() ? 1 : 0,
+        message: this.serviceStatusMessage(),
+      },
+    };
+  }
+
+  private serviceStatusMessage() {
+    switch (this.connection.readyState) {
+      case 0 || 2 || 3:
```

Comment:
:o Never seen this signature, is it working?

Reply:
Changed to something better to case stack.
case 0:
case 2:
case 3: